### PR TITLE
Propogate etcd preflight network errors upstream #50622 [WIP]

### LINKED ISSUE #50622: apiserver errors should include reason
/kind feature

**What happened**:
Spend too much time figuring out that my apiserver didn't start because the etcd DNS name couldn't be resolved. I got the following error:
```
Error: error waiting for etcd connection: timed out waiting for the condition
```
And even `--v=10` didn't provide any more context. This was very misleading and made me assume the issue is on application level.

**What you expected to happen**:
I expected the apiserver error to tell me that DNS resolution wasn't working.

**How to reproduce it (as minimally and precisely as possible)**:
I used the bootkube manifest but just starting the apiserver with `--etcd-servers` pointing to a nonexisting name.

### PR DESCRIPTION
**Pull request by jordy25519**
**What this PR does / why we need it**:
Errors from pre-flight checks on etcd server connections are masked by the API server as: "`timed out waiting for the condition`". This PR ensures the network error is surfaced to aid debugging configuration etc.

**Which issue(s) this PR fixes** *
Fixes #50622 

**Special notes for your reviewer**:

**Release note**:

```release-note
Improve etcd connection error logging on api server start-up
```

### CODE REVIEW

**Comment by holygits** [comment]
apologies for multiple retests, I rebased and think that introduced some unexpected test failures

**Comment by Cryptophobia** [comment]
This very good. Are all errors raised up the stack every time?

File: staging/src/k8s.io/apiserver/pkg/server/options/etcd.go [hunk 1]
```diff
@@ -197,7 +197,7 @@ func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) {
 	c.HealthzChecks = append(c.HealthzChecks, healthz.NamedCheck("etcd", func(r *http.Request) error {
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/server/options/etcd.go]: we're switching the etcd health check to make an actual etcd request with a real client in https://github.com/kubernetes/kubernetes/pull/65027 to address #64909

**Comment by Cryptophobia** [comment]
/remove-lifecycle stale

Would this be ready for review since https://github.com/kubernetes/kubernetes/pull/65027 has been merged?

**Comment by holygits** [comment]
AFAIK I'll need to update this based on changes in #65027

**Comment by holygits** [comment]
Closing in favour of #65027

**Pull request closed without merging**